airflow-core/tests/unit/models/test_taskinstance.py

```diff
@@ -1685,7 +1685,9 @@ def _write2_post_execute(context, _):
         for ti in dr.get_task_instances(session=session):
             ti.run(session=session)
 
-        events = dict(iter(session.execute(select(AssetEvent.source_task_id, AssetEvent))))
+        events = dict(
+            (row[0], row[1]) for row in session.execute(select(AssetEvent.source_task_id, AssetEvent))
```

Comment:
Perhaps the below is safer? Consider a scenario where the produced row has more than 2 elements for some reason, the test will not capture that because we only look at the first two. If on the other hand we convert the entire row, the test would fail, as it should.

```suggestion
            (tuple(row)) for row in session.execute(select(AssetEvent.source_task_id, AssetEvent))
```

Reply:
Yes, agree! Thanks for review and comments! :)